src/GafferScene/SceneAlgo.cpp

```diff
@@ -127,6 +127,22 @@ struct ThreadablePathAccumulator
 
 };
 
+struct ThreadablePathHashAccumulator
+{
+	ThreadablePathHashAccumulator(): m_h1Accum( 0 ), m_h2Accum( 0 ){}
+
+	bool operator()( const GafferScene::ScenePlug *scene, const GafferScene::ScenePlug::ScenePath &path )
+	{
+		IECore::MurmurHash h;
+		h.append( &path.front(), path.size() );
+		m_h1Accum += h.h1();
+		m_h2Accum += h.h2();
+		return true;
+	}
+
+	std::atomic<uint64_t> m_h1Accum, m_h2Accum;
```

Comment:
I also experimented using `enumerable_thread_specific` here, and again it was inconclusive : https://github.com/johnhaddon/gaffer/commit/7adc862518c3b29718111d49d1a3ed87e29436cb.